### ISSUE TITLE
[Feature] An option to silence warnings

### ISSUE THREAD

**Issue by jtcohen6** [dbt-labs/dbt-core#9644]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

https://docs.getdbt.com/reference/global-configs/warnings

```yml
warn_error_options:
  error: # previously called "include", promote to "error"
  warn: # previously called "exclude", keep as "warn"
  silence: # NEW: ignore entirely
```

(We'd retain the old names as well for backwards compatability)

<img width="270" alt="Screenshot 2024-02-23 at 10 56 05" src="https://github.com/dbt-labs/dbt-core/assets/13897643/94b3de74-7c68-4293-89e1-c40ce8770842">

### Describe alternatives you've considered

- A separate config, `WARN_SILENCE`, that takes a list. This would be better in the case where a user wants to `silence` warnings via project flags (version-controlled default for everyone), and also promote specific warnings as errors (via `include`/`exclude`) in different runtime environments, without needing to re-specify the `silence` list every time. While that's slightly more ergonomic, it's also less explicit, and it doesn't feel like a show-stopper.
- Only allow users to silence **deprecation** warnings. I think there's a valid reason to generalize this to all kinds of warnings, e.g. silence `ConstraintNotEnforced` once for the entire project rather than once per model/constraint in the project.
- Remove `TestsConfigDeprecation` (= give up in advance)

### Who will this benefit?

All users of dbt, especially since:
- In v1.8, we are renaming `tests` &rarr; to `data_tests`, for clearer distinction from `unit_tests`. Tens of thousands of projects will experience the `TestsConfigDeprecation`. They won't all be able to take action. We should give them a way to silence the deprecation (for now).
- We should still reword that deprecation message! https://github.com/dbt-labs/dbt-core/issues/9579

### Are you interested in contributing this feature?

:)

### Anything else?

Sketched out a start:
- https://github.com/dbt-labs/dbt-common/commit/60a6ef125de49d98609550cd1ed3ba2a08ed71ea
- https://github.com/dbt-labs/dbt-core/commit/31b10023c7e56f2cc922e429c671485e9c8f1de4

Example:
```yml
$ cat dbt_project.yml
name: "my_dbt_project"

tests:
  +enabled: True

flags:
  warn_error_options:
    silence:
      - TestsConfigDeprecation
```

```
$ dbt --no-partial-parse parse
16:02:55  Running with dbt=1.8.0-a1
16:02:56  Registered adapter: duckdb=1.7.0
16:02:57  Performance info: /Users/jerco/dev/scratch/testy/target/perf_info.json
```

[Slack thread with more context](https://dbt-labs.slack.com/archives/C03SK9W08BC/p1708704701904679)

**Comment by MichelleArk** [comment]
From technical refinment: 
* would need to ensure all instantiations of the WarnErrorOptions support both `include/exclude` and `warn/error` syntax
  *. eg: CLI instantiation: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/cli/option_types.py#L56-L61

**Comment by martynydbt** [comment]
@QMalcolm I removed the blocker since you're unblocked.

**Comment by FishtownBuildBot** [comment]
Opened a new issue in dbt-labs/docs.getdbt.com: https://github.com/dbt-labs/docs.getdbt.com/issues/5417

**Comment by mirnawong1** [comment]
we've opened a docs PR https://github.com/dbt-labs/docs.getdbt.com/pull/5434 and sharing the [preview](https://docs-getdbt-dywzt6k2v-dbt-labs.vercel.app/reference/global-configs/warnings)
